bitcoin/tests/psbt.rs

```diff
@@ -26,10 +26,10 @@ macro_rules! hex_script {
     };
 }
 
-macro_rules! hex_psbt {
-    ($s:expr) => {
-        Psbt::deserialize(&<Vec<u8> as FromHex>::from_hex($s).unwrap())
-    };
+#[track_caller]
+pub fn hex_psbt(s: &str) -> Psbt {
+    let v: Vec<u8> = Vec::from_hex(s).unwrap();
+    Psbt::deserialize(&v[..]).unwrap()
```

Comment:
We don't need type annotation here, and don't need the explicit slicing. I don't see a reason not to use `expect` to quickly/easily differentiate between the two `unwraps` during debugging. With all that perhaps this?
```rust
#[track_caller]
pub fn hex_psbt(s: &str) -> Psbt {
    let v = Vec::from_hex(s).expect("failed to parse hex");
    Psbt::deserialize(&v).expect("failed to deserialize vec")
}
```

Reply:
Thanks, I've added `expect()`

Reply:
Your example expect() messages are not idiomatic enough :).  It seems counter intuitive sort of, but expect messages should state what is expected, not that it's expected to fail.  For example, checkout the expect messages of https://docs.rs/hex-conservative/latest/hex_conservative/.

Reply:
updated to:

```rust
#[track_caller]
fn hex_psbt(s: &str) -> Psbt {
    let v: Vec<u8> = Vec::from_hex(s).expect("valid hex digits");
    Psbt::deserialize(&v).expect("valid magic and valid separators")
}
```

Reply:
ah yes, you are correct. I routinely make this mistake.